packages/api/src/Api.ts

```diff
@@ -31,9 +31,26 @@ import logger from './util/logging';
 
 const Types = require('@cennznet/types');
 
+export const DEFAULT_TIMEOUT = 30000;
+
 export class Api extends ApiPromise {
-    static async create(options: ApiOptions | ProviderInterface = {}): Promise<Api> {
-        return new Api(options).isReady;
+    static async create(provider: ApiOptions | ProviderInterface = {}): Promise<Api> {
+        const options =
+            isObject(provider) && isFunction((provider as ProviderInterface).send)
+                ? ({provider} as ApiOptions)
+                : ({...provider} as ApiOptions);
+
+        const api = await new Api(options);
+        return withTimeout(
+            new Promise((resolve, reject) => {
+                api.isReady.then(api => resolve(api)).catch(e => reject(e));
```

Comment:
api.isReady.then(resolve, reject);

Reply:
resolve(api.isReady) should do the same thing